peachjam/views/legislation.py

```diff
@@ -17,3 +22,77 @@ class LegislationListView(FilteredDocumentListView):
 class LegislationDetailView(BaseDocumentDetailView):
     model = Legislation
     template_name = "peachjam/legislation_detail.html"
+
+    def get_notices(self):
+        notices = super().get_notices()
+        repeal = self.get_repeal_info()
+        if self.object.repealed and repeal:
+            msg = "This {} was repealed on {} by <a href='{}'>{}</a>"
+            notices.append(
+                {
+                    "type": messages.ERROR,
+                    "html": format_html(
+                        msg.format(
+                            self.object.metadata_json["type_name"],
+                            repeal["date"],
+                            repeal["repealing_uri"],
+                            repeal["repealing_title"],
+                        )
+                    ),
+                }
+            )
+
+            msg = (
+                "This is the latest version of this {} as it was when it was repealed."
+            )
+            notices.append(
+                {
+                    "type": messages.INFO,
+                    "html": format_html(
+                        msg.format(
+                            self.object.metadata_json["type_name"],
+                        )
+                    ),
+                }
+            )
+
+        # fetch the points in time
+        points_in_time = self.get_points_in_time()
+        latest_expression = points_in_time[-1]
+
+        for idx, point_in_time in enumerate(points_in_time):
+            if point_in_time != latest_expression:
```

Comment:
This case seems to be missing: For the most up-to-date version of amended legislation, the notice should say "This is the latest version of this (friendly type)".